iree/test/e2e/xla_ops/partial/add_f16.mlir

```diff
@@ -0,0 +1,7 @@
+func @tensor() attributes { iree.module.export } {
+  %0 = iree.unfoldable_constant dense<[1.0, 2.0, 3.0, 4.0]> : tensor<4xf16>
+  %1 = iree.unfoldable_constant dense<[5.0, 6.0, 7.0, 8.0]> : tensor<4xf16>
+  %result = "mhlo.add"(%0, %1) : (tensor<4xf16>, tensor<4xf16>) -> tensor<4xf16>
+  check.expect_almost_eq_const(%result, dense<[6.0, 8.0, 10.0, 12.0]> : tensor<4xf16>) : tensor<4xf16>
```

Comment:
Eh.. this is tricky. We _could_ (and probably will eventually) correctly execute these tests without native support for 16 bit floating point numbers by emulating in `f32` or widening. Native support is just an optimization. I still see the point of these tests, but we should be careful with how we categorize them.

Reply:
Using f32 to emulate would work but we would still need some conversion to and from float 16 for data coming from user, this is what may make the support a bit more complex. But using float32 for computation should be valid indeed.

Reply:
Yeah I would hope that eventually these tests get put somewhere common to all backends. Eventually mhlo may not even be our input, so we'll probably change the way everything is organized (falls under our integration tests roadmap OKR for next quarter)